components/k8s-dashboard-proxy/cmd/reverseproxy/reverseproxy.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+)
+
+const (
+	dashboardURL   = "http://localhost:30000"
+	secretFilePath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	port           = ":8080"
+)
+
+var (
+	proxy    *httputil.ReverseProxy
+	proxyURL *url.URL
+)
+
+func readSecret() string {
+	var accountSecret []byte
+	if token, err := ioutil.ReadFile(secretFilePath); err != nil {
+		log.Printf("Error: read service account failed: %v", err)
+		accountSecret = token
+		return ""
+	}
+	return string(accountSecret)
+}
+
+func serveProxy(rw http.ResponseWriter, req *http.Request) {
+	req.URL.Host = proxyURL.Host
+	req.URL.Scheme = proxyURL.Scheme
+	req.Header.Set("Authorization", "Bearer "+readSecret())
```

Comment:
Please try to define a kind of "cache" here, so that the  "secret" should be not read at each request. This "caching time" should be made configurable.

Reply:
Yeah, we had this planned already, but I'd say moving this as a separate optimization task to unlock this one, what do you think?
CC: @Abd4llA

Reply:
Th idea is to use something very simple, like "time.NewTicker()" with a default interval (perhaps 60s).